src/Testcontainers.CosmosDb/CosmosDbContainer.cs

```diff
@@ -20,11 +22,23 @@ public CosmosDbContainer(CosmosDbConfiguration configuration)
     public string GetConnectionString()
     {
         var properties = new Dictionary<string, string>();
-        properties.Add("AccountEndpoint", new UriBuilder(Uri.UriSchemeHttps, Hostname, GetMappedPublicPort(CosmosDbBuilder.CosmosDbPort)).ToString());
+        properties.Add("AccountEndpoint", new UriBuilder(Uri.UriSchemeHttp, Hostname, GetMappedPublicPort(CosmosDbBuilder.CosmosDbPort)).ToString());
         properties.Add("AccountKey", CosmosDbBuilder.DefaultAccountKey);
         return string.Join(";", properties.Select(property => string.Join("=", property.Key, property.Value)));
     }
 
+    /// <summary>
+    /// Gets a configured cosmos client with connection mode set to Gateway.
+    /// </summary>
+    public CosmosClient CosmosClient
+        => new CosmosClient(
+            GetConnectionString(),
+            new()
+            {
+                ConnectionMode = ConnectionMode.Gateway,
+                HttpClientFactory = () => new(new UriRewriter(Hostname, GetMappedPublicPort(CosmosDbBuilder.CosmosDbPort)))
```

Comment:
You can use this.HttpClient here.

Reply:
According to the discussion [here](https://github.com/testcontainers/testcontainers-dotnet/discussions/1306#discussioncomment-11745647), it was agreed that it would be better to remove the dependency on the Cosmos Client and delegate the responsibility of creating the client to the consumer.